aten/src/ATen/functorch/BatchRulesBinaryOps.cpp

```diff
@@ -287,6 +287,24 @@ void fill__Tensor_batch_rule(
   std::get<0>(self_and_other).copy_(std::get<1>(self_and_other));
 }
 
+std::tuple<Tensor,optional<int64_t>> fill_Tensor_batch_rule(
+    Tensor& self,
+    optional<int64_t> self_bdim,
+    const Tensor& other,
+    optional<int64_t> other_bdim) {
+  if (!other_bdim.has_value()) {
+    // Optimization: fill is faster than the other path which does
+    // reshaping + copy_
+    return std::make_tuple(at::fill(self, other), self_bdim);
+  }
+  auto self_and_other = _binary_pointwise_helper(
```

Comment:
Since this path doesn't call fill, do we need to do repeat any of the shape checks that fill would normally do? Specifically if other is a matrix (batched vector) instead of a vector (batched scalar), won't this silently pass instead of producing an error?

Reply:
You're right, there needs to be an error check somewhere